Src/Support/GoogleApis.Auth.PlatformServices_Shared/OAuth2/Pkcs8.cs

```diff
@@ -0,0 +1,266 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using Google.Apis.Util;
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    internal class Pkcs8
+    {
+        // PKCS#8 specification: https://www.ietf.org/rfc/rfc5208.txt
+        // ASN.1 specification: https://www.itu.int/ITU-T/studygroups/com17/languages/X.690-0207.pdf
+
+        /// <summary>
+        /// An incomplete ASN.1 decoder, only implements what's required
+        /// to decode a Service Credential.
+        /// </summary>
+        internal class Asn1
+        {
+            internal enum Tag
+            {
+                Integer = 2,
+                OctetString = 4,
+                Null = 5,
+                ObjectIdentifier = 6,
+                Sequence = 16,
+            }
+
+            internal class Decoder
+            {
+                public Decoder(byte[] bs)
+                {
+                    _bs = bs;
+                    _index = 0;
+                }
+
+                private byte[] _bs;
+                private int _index;
+
+                public object Decode()
+                {
+                    Tag tag = ReadTag();
+                    switch (tag)
+                    {
+                        case Tag.Integer:
+                            return ReadInteger();
+                        case Tag.OctetString:
+                            return ReadOctetString();
+                        case Tag.Null:
+                            return ReadNull();
+                        case Tag.ObjectIdentifier:
+                            return ReadOid();
+                        case Tag.Sequence:
+                            return ReadSequence();
+                        default:
+                            throw new NotSupportedException($"Tag '{tag}' not supported.");
+                    }
+                }
+
+                private byte[] ReadInteger()
+                {
+                    int length = ReadLength();
+                    return ReadBytes(length);
+                }
+
+                private object ReadOctetString()
+                {
+                    int length = ReadLength();
+                    byte[] bs = ReadBytes(length);
+                    return new Decoder(bs).Decode();
+                }
+
+                private object ReadNull()
+                {
+                    int length = ReadLength();
+                    if (length != 0)
+                    {
+                        throw new InvalidOperationException("Invalid data, Null length must be 0.");
+                    }
+                    return null;
+                }
+
+                private int[] ReadOid()
+                {
+                    int length = ReadLength();
```

Comment:
Is there any reason this doesn't call ReadBytes, but reads one byte at a time instead?